# Introduction

The data was gathered around May 2019, utilizing the Steam Store and SteamSpy APIs, as stated by the dataset's author on the description.

This is an exploratory analysis of this dataset, mainly utilizing Pandas and its functions. I will try to answer the following questions:

- Top 20 games with the most positive ratings
- Top 20 games with the most negative ratings
- Top 20 games with the highest average playtime
- Price analysis
- Distribution of genres
- Publishers with the highest number of games
- Games released by year
- What does ownership look like

In [2]:
import pandas as pd
import numpy as np

## **Initial Dataset Examination**

In [5]:
store = pd.read_csv('steam.csv')
store

appid                        name release_date  english  \
0           10              Counter-Strike   2000-11-01        1   
1           20       Team Fortress Classic   1999-04-01        1   
2           30               Day of Defeat   2003-05-01        1   
3           40          Deathmatch Classic   2001-06-01        1   
4           50   Half-Life: Opposing Force   1999-11-01        1   
...        ...                         ...          ...      ...   
27070  1065230             Room of Pandora   2019-04-24        1   
27071  1065570                   Cyber Gun   2019-04-23        1   
27072  1065650            Super Star Blast   2019-04-24        1   
27073  1066700  New Yankee 7: Deer Hunters   2019-04-17        1   
27074  1069460                   Rune Lord   2019-04-24        1   

                developer             publisher          platforms  \
0                   Valve                 Valve  windows;mac;linux   
1                   Valve                 Valve  windows;mac;linux   
2                   Valve                 Valve  windows;mac;linux   
3                   Valve                 Valve  windows;mac;linux   
4        Gearbox Software                 Valve  windows;mac;linux   
...                   ...                   ...                ...   
27070         SHEN JIAWEI           SHEN JIAWEI            windows   
27071      Semyon Maximov      BekkerDev Studio            windows   
27072         EntwicklerX           EntwicklerX            windows   
27073  Yustas Game Studio  Alawar Entertainment        windows;mac   
27074    Adept Studios GD  Alawar Entertainment        windows;mac   

       required_age                                         categories  \
0                 0  Multi-player;Online Multi-Player;Local Multi-P...   
1                 0  Multi-player;Online Multi-Player;Local Multi-P...   
2                 0              Multi-player;Valve Anti-Cheat enabled   
3                 0  Multi-player;Online Multi-Player;Local Multi-P...   
4                 0  Single-player;Multi-player;Valve Anti-Cheat en...   
...             ...                                                ...   
27070             0                   Single-player;Steam Achievements   
27071             0                                      Single-player   
27072             0  Single-player;Multi-player;Co-op;Shared/Split ...   
27073             0                          Single-player;Steam Cloud   
27074             0                          Single-player;Steam Cloud   

                       genres                 steamspy_tags  achievements  \
0                      Action        Action;FPS;Multiplayer             0   
1                      Action        Action;FPS;Multiplayer             0   
2                      Action  FPS;World War II;Multiplayer             0   
3                      Action        Action;FPS;Multiplayer             0   
4                      Action             FPS;Action;Sci-fi             0   
...                       ...                           ...           ...   
27070  Adventure;Casual;Indie        Adventure;Indie;Casual             7   
27071  Action;Adventure;Indie        Action;Indie;Adventure             0   
27072     Action;Casual;Indie           Action;Indie;Casual            24   
27073  Adventure;Casual;Indie        Indie;Casual;Adventure             0   
27074  Adventure;Casual;Indie        Indie;Casual;Adventure             0   

       positive_ratings  negative_ratings  average_playtime  median_playtime  \
0                124534              3339             17612              317   
1                  3318               633               277               62   
2                  3416               398               187               34   
3                  1273               267               258              184   
4                  5250               288               624              415   
...                 ...               ...               .

In [7]:
store.columns

Index(['appid', 'name', 'release_date', 'english', 'developer', 'publisher',
       'platforms', 'required_age', 'categories', 'genres', 'steamspy_tags',
       'achievements', 'positive_ratings', 'negative_ratings',
       'average_playtime', 'median_playtime', 'owners', 'price'],
      dtype='object')

Getting a summary of the statistics in my dataset

In [8]:
store.describe()

appid       english  required_age  achievements  \
count  2.707500e+04  27075.000000  27075.000000  27075.000000   
mean   5.962035e+05      0.981127      0.354903     45.248864   
std    2.508942e+05      0.136081      2.406044    352.670281   
min    1.000000e+01      0.000000      0.000000      0.000000   
25%    4.012300e+05      1.000000      0.000000      0.000000   
50%    5.990700e+05      1.000000      0.000000      7.000000   
75%    7.987600e+05      1.000000      0.000000     23.000000   
max    1.069460e+06      1.000000     18.000000   9821.000000   

       positive_ratings  negative_ratings  average_playtime  median_playtime  \
count      2.707500e+04      27075.000000      27075.000000      27075.00000   
mean       1.000559e+03        211.027147        149.804949        146.05603   
std        1.898872e+04       4284.938531       1827.038141       2353.88008   
min        0.000000e+00          0.000000          0.000000          0.00000   
25%        6.000000e+00          2.000000          0.000000          0.00000   
50%        2.400000e+01          9.000000          0.000000          0.00000   
75%        1.260000e+02         42.000000          0.000000          0.00000   
max        2.644404e+06     487076.000000     190625.000000     190625.00000   

              price  
count  27075.000000  
mean       6.078193  
std        7.874922  
min        0.000000  
25%        1.690000  
50%        3.990000  
75%        7.190000  
max      421.990000

In [10]:
store.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27075 entries, 0 to 27074
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   appid             27075 non-null  int64  
 1   name              27075 non-null  object 
 2   release_date      27075 non-null  object 
 3   english           27075 non-null  int64  
 4   developer         27074 non-null  object 
 5   publisher         27061 non-null  object 
 6   platforms         27075 non-null  object 
 7   required_age      27075 non-null  int64  
 8   categories        27075 non-null  object 
 9   genres            27075 non-null  object 
 10  steamspy_tags     27075 non-null  object 
 11  achievements      27075 non-null  int64  
 12  positive_ratings  27075 non-null  int64  
 13  negative_ratings  27075 non-null  int64  
 14  average_playtime  27075 non-null  int64  
 15  median_playtime   27075 non-null  int64  
 16  owners            27075 non-null  object

Verifying how many lines and columns my dataset has

In [9]:
store.shape

(27075, 18)

Checking if there are any null values in my dataframe. This is good practice, although this dataset was already cleaned by the author.

In [11]:
store.isnull().sum()

appid                0
name                 0
release_date         0
english              0
developer            1
publisher           14
platforms            0
required_age         0
categories           0
genres               0
steamspy_tags        0
achievements         0
positive_ratings     0
negative_ratings     0
average_playtime     0
median_playtime      0
owners               0
price                0
dtype: int64

Columns appid and steamspy_tags will be dropped. The former because it doesn't have any relevant information for this exploratory analysis, and the latter because it has very similar data to the genres column, and that is the chosen column to be used in this analysis.

In [12]:

store.drop(columns=['appid', 'steamspy_tags'], inplace=True)
store.columns




Index(['name', 'release_date', 'english', 'developer', 'publisher',
       'platforms', 'required_age', 'categories', 'genres', 'achievements',
       'positive_ratings', 'negative_ratings', 'average_playtime',
       'median_playtime', 'owners', 'price'],
      dtype='object')

Altering the genres column inplace. Since this column has many genres separated by semicolons, we will disregard the ones that come after the first and consider that the main genre.

In [15]:

store['genres'] = store['genres'].apply(lambda x: x.split(';')[0] if isinstance(x, str) else x)

store[['genres']]


genres
0         Action
1         Action
2         Action
3         Action
4         Action
...          ...
27070  Adventure
27071     Action
27072     Action
27073  Adventure
27074  Adventure

[27075 rows x 1 columns]

There's a lot of zeroes in the column required_age, so a query is run to check whether that is the situation for the entire column or not, in case it's bad data

In [17]:

all_zeroes = (store['required_age'] == 0).all()

if all_zeroes:
    print("The entire 'required_age' column consists of zeros.")
else:
    print("The 'required_age' column contains non-zero values as well.")


store['required_age'].unique()


The 'required_age' column contains non-zero values as well.


array([ 0, 18, 16, 12,  7,  3])

In [18]:
store.head()

name release_date  english         developer  \
0             Counter-Strike   2000-11-01        1             Valve   
1      Team Fortress Classic   1999-04-01        1             Valve   
2              Day of Defeat   2003-05-01        1             Valve   
3         Deathmatch Classic   2001-06-01        1             Valve   
4  Half-Life: Opposing Force   1999-11-01        1  Gearbox Software   

  publisher          platforms  required_age  \
0     Valve  windows;mac;linux             0   
1     Valve  windows;mac;linux             0   
2     Valve  windows;mac;linux             0   
3     Valve  windows;mac;linux             0   
4     Valve  windows;mac;linux             0   

                                          categories  genres  achievements  \
0  Multi-player;Online Multi-Player;Local Multi-P...  Action             0   
1  Multi-player;Online Multi-Player;Local Multi-P...  Action             0   
2              Multi-player;Valve Anti-Cheat enabled  Action             0   
3  Multi-player;Online Multi-Player;Local Multi-P...  Action             0   
4  Single-player;Multi-player;Valve Anti-Cheat en...  Action             0   

   positive_ratings  negative_ratings  average_playtime  median_playtime  \
0            124534              3339             17612              317   
1              3318               633               277               62   
2              3416               398               187               34   
3              1273               267               258              184   
4              5250               288               624              415   

              owners  price  
0  10000000-20000000   7.19  
1   5000000-10000000   3.99  
2   5000000-10000000   3.99  
3   5000000-10000000   3.99  
4   5000000-10000000   3.99

## **Top 20 Games with the Most Positive Ratings**

Discovering the top 20 games with the most positive reviews on Steam. First, the 20 largest values in the positive_ratings column will be filtered.

In [19]:
top_20_positive = store.sort_values(by='positive_ratings', ascending=False).head(20)[['name', 'positive_ratings']]
top_20_positive


name  positive_ratings
25     Counter-Strike: Global Offensive           2644404
22                               Dota 2            863507
19                      Team Fortress 2            515879
12836     PLAYERUNKNOWN'S BATTLEGROUNDS            496184
121                         Garry's Mod            363721
2478                 Grand Theft Auto V            329061
1467                           PAYDAY 2            308657
3362                           Unturned            292574
1120                           Terraria            255600
21                        Left 4 Dead 2            251789
5235    Tom Clancy's Rainbow Six® Siege            251178
2031                     Rocket League®            242561
1025        The Elder Scrolls V: Skyrim            237303
1634                           Warframe            226541
2016                               Rust            220370
2964          The Witcher® 3: Wild Hunt            202930
1596             Euro Truck Simulator 2            176769
8129                          Paladins®            169580
4712              ARK: Survival Evolved            145035
903                       Borderlands 2            144595

Then, a query is applied to select only those games whose quantity of positive ratings is above the last number retrieved in the filter. In this case, that would be 144595.

In [27]:
# Get top 20 games by positive ratings
top_20_positive = store.nlargest(20, 'positive_ratings')

# Get the minimum positive rating among the top 20
threshold = top_20_positive['positive_ratings'].min()

# Select all games with positive ratings greater than or equal to the threshold
all_above_threshold = store[store['positive_ratings'] >= threshold].sort_values(by='positive_ratings', ascending=False)


print(f"Number of games with positive ratings >= {threshold}: {len(all_above_threshold)}")
all_above_threshold[['name', 'positive_ratings']]



Number of games with positive ratings >= 144595: 20


name  positive_ratings
25     Counter-Strike: Global Offensive           2644404
22                               Dota 2            863507
19                      Team Fortress 2            515879
12836     PLAYERUNKNOWN'S BATTLEGROUNDS            496184
121                         Garry's Mod            363721
2478                 Grand Theft Auto V            329061
1467                           PAYDAY 2            308657
3362                           Unturned            292574
1120                           Terraria            255600
21                        Left 4 Dead 2            251789
5235    Tom Clancy's Rainbow Six® Siege            251178
2031                     Rocket League®            242561
1025        The Elder Scrolls V: Skyrim            237303
1634                           Warframe            226541
2016                               Rust            220370
2964          The Witcher® 3: Wild Hunt            202930
1596             Euro Truck Simulator 2            176769
8129                          Paladins®            169580
4712              ARK: Survival Evolved            145035
903                       Borderlands 2            144595

Unsurprisingly, the podium is occupied by the wildly popular PC games Counter-Strike: Global Offensive, Dota 2 and Team Fortress 2, all three developed by Steam's own parent company, Valve Software.

## **Top 20 Games with the Most Negative Ratings**

Now, going to the opposite end, discovering the top 20 games with the most negative reviews on Steam. Filtering for the the 20 largest values in the negative_ratings column.

In [ ]:
top_20_negative_simple = store.sort_values(by='negative_ratings', ascending=False).head(20)[['name', 'negative_ratings']]
print(top_20_negative_simple)


name  negative_ratings
12836     PLAYERUNKNOWN'S BATTLEGROUNDS            487076
25     Counter-Strike: Global Offensive            402313
22                               Dota 2            142079
2478                 Grand Theft Auto V            139308
7809                   Z1 Battle Royale             91664
1506                               DayZ             77169
4712              ARK: Survival Evolved             66603
5235    Tom Clancy's Rainbow Six® Siege             59620
1467                           PAYDAY 2             56523
2563                       No Man's Sky             56488
2016                               Rust             53709
5916                          Fallout 4             45377
6046                   Dead by Daylight             44985
12965             MONSTER HUNTER: WORLD             35294
1607                  Heroes & Generals             34416
19                      Team Fortress 2             34036
8129                          Paladins®             32407
3362                           Unturned             31482
3251                          Robocraft             30987
3347                         FOR HONOR™             25920

Applying a query to select only those games whose quantity of negative ratings is above the last number retrieved in the filter, 25920.

In [26]:

top_20_negative = store.nlargest(20, 'negative_ratings')

threshold = top_20_negative['negative_ratings'].min()

all_above_threshold = store[store['negative_ratings'] >= threshold].sort_values(by='negative_ratings', ascending=False)

print(f"Number of games with negative ratings >= {threshold}: {len(all_above_threshold)}")
all_above_threshold[['name', 'negative_ratings']]


Number of games with negative ratings >= 25920: 20


name  negative_ratings
12836     PLAYERUNKNOWN'S BATTLEGROUNDS            487076
25     Counter-Strike: Global Offensive            402313
22                               Dota 2            142079
2478                 Grand Theft Auto V            139308
7809                   Z1 Battle Royale             91664
1506                               DayZ             77169
4712              ARK: Survival Evolved             66603
5235    Tom Clancy's Rainbow Six® Siege             59620
1467                           PAYDAY 2             56523
2563                       No Man's Sky             56488
2016                               Rust             53709
5916                          Fallout 4             45377
6046                   Dead by Daylight             44985
12965             MONSTER HUNTER: WORLD             35294
1607                  Heroes & Generals             34416
19                      Team Fortress 2             34036
8129                          Paladins®             32407
3362                           Unturned             31482
3251                          Robocraft             30987
3347                         FOR HONOR™             25920

Popular video games, especially those with live-service and multiplayer components, can often be very divisive or controversial due to the fickle nature of the player bases and the state of their satisfaction or dissatisfaction with the games as they get updated and changed. Thus Counter-Strike: Global Offensive and Dota 2 occupy 2 of the 3 spots of games with the most negative reviews.

## **Top 20 Games with the Highest Average Playtime**

Taking a look at how much time people spend on Steam's most popular games, on average.

First, the average_playtime column will be converted from minutes to hours for better visualization

In [28]:
# change time from minutes to hours
store['average_playtime_hours'] = store['average_playtime'] / 60


top_20_playtime = store.sort_values(by='average_playtime_hours', ascending=False).head(20)


top_20_playtime = top_20_playtime[['name', 'average_playtime_hours']]

top_20_playtime


name  average_playtime_hours
9201               The Abbey of Crime Extensum             3177.083333
1478                 The Banner Saga: Factions             1587.416667
6014      The Secret of Tremendous Corporation             1587.366667
8969                                     PRICE             1058.016667
3930                                 Boundless              920.066667
3969    Shroud of the Avatar: Forsaken Virtues              910.300000
2435                                X-Plane 11              736.150000
23409                                    懒人修仙传              727.200000
2023                           Fantasy Grounds              717.900000
8919                                   Screeps              646.750000
12195                     The Price of Freedom              600.483333
744                   FINAL FANTASY XIV Online              481.616667
3803                                 Darkstone              456.250000
15212                               Warp Rider              422.066667
16413   Trivia Vault: Science & History Trivia              422.033333
16143                           The Moon Night              421.950000
22                                      Dota 2              399.066667
12836            PLAYERUNKNOWN'S BATTLEGROUNDS              382.300000
25            Counter-Strike: Global Offensive              374.900000
2737   Heroine's Quest: The Herald of Ragnarok              354.116667

## **Price Analysis**

A simple price analysis will be shown below, focusing on the split of free vs. paid games, average price by year and general price distribution. 

release_year and release_month columns will be created from the release_date one, as it will be used for this price analysis and a few other metrics down the line

In [31]:

store['release_date'] = pd.to_datetime(store['release_date'], errors='coerce')


store['release_year'] = store['release_date'].dt.year
store['release_month'] = store['release_date'].dt.month

free_vs_paid = store['price'].apply(lambda x: 'Free' if x == 0 else 'Paid').value_counts()

print("Free vs Paid games count:")
print(free_vs_paid)

avg_price_by_year = store[store['price'] > 0].groupby('release_year')['price'].mean().dropna()

print("\nAverage price by release year:")
print(avg_price_by_year)

price_distribution = store['price'].describe()

print("\nPrice distribution summary:")
print(price_distribution)


Free vs Paid games count:
price
Paid    24515
Free     2560
Name: count, dtype: int64

Average price by release year:
release_year
1997    5.990000
1998    7.190000
1999    3.990000
2000    5.590000
2001    3.990000
2002    7.990000
2003    4.490000
2004    6.910000
2005    6.040000
2006    8.849024
2007    6.021494
2008    6.916861
2009    7.227356
2010    7.645870
2011    8.067937
2012    9.284932
2013    9.367010
2014    7.878242
2015    6.920021
2016    6.553470
2017    6.505873
2018    6.219752
2019    7.077412
Name: price, dtype: float64

Price distribution summary:
count    27075.000000
mean         6.078193
std          7.874922
min          0.000000
25%          1.690000
50%          3.990000
75%          7.190000
max        421.990000
Name: price, dtype: float64


A new boolean column will be added called free, this column displays whether a game is free or paid

In [ ]:

store['free'] = store['price'] == 0

print(store[['name', 'price', 'free']].head(10))


                             name  price   free
0                  Counter-Strike   7.19  False
1           Team Fortress Classic   3.99  False
2                   Day of Defeat   3.99  False
3              Deathmatch Classic   3.99  False
4       Half-Life: Opposing Force   3.99  False
5                        Ricochet   3.99  False
6                       Half-Life   7.19  False
7  Counter-Strike: Condition Zero   7.19  False
8           Half-Life: Blue Shift   3.99  False
9                     Half-Life 2   7.19  False


## **Publishers and Developers with the Highest Number of Games**

In [ ]:

top_publishers = store['publisher'].value_counts().head(10)

top_publishers


publisher
Big Fish Games         212
Strategy First         136
Ubisoft                111
THQ Nordic              98
Square Enix             97
Sekai Project           96
Choice of Games         94
1C Entertainment        88
Dagestan Technology     88
SEGA                    78
Name: count, dtype: int64

In [40]:

top_developers = store['developer'].value_counts().head(10)

top_developers


developer
Choice of Games               94
KOEI TECMO GAMES CO., LTD.    72
Ripknot Systems               62
Laush Dmitriy Sergeevich      51
Nikita "Ghost_RUS"            50
Dexion Games                  45
RewindApp                     43
Hosted Games                  42
Blender Games                 40
Humongous Entertainment       36
Name: count, dtype: int64

Top 10 games by ownership

In [41]:

store['min_owners'] = store['owners'].str.split('-').str[0].astype(int)


top_10_games_by_ownership = store.sort_values(by='min_owners', ascending=False).head(10)


top_10_games_by_ownership[['name', 'owners']]


name               owners
22                               Dota 2  100000000-200000000
12836     PLAYERUNKNOWN'S BATTLEGROUNDS   50000000-100000000
25     Counter-Strike: Global Offensive   50000000-100000000
19                      Team Fortress 2    20000000-50000000
3362                           Unturned    20000000-50000000
1634                           Warframe    20000000-50000000
0                        Counter-Strike    10000000-20000000
23                             Portal 2    10000000-20000000
7809                   Z1 Battle Royale    10000000-20000000
8129                          Paladins®    10000000-20000000